Makefile

```diff
@@ -89,22 +89,14 @@ docker-build:
 docker-push:
 	docker push ${IMG}
 
-# find or download controller-gen
-# download controller-gen if necessary
-controller-gen:
-ifeq (, $(shell which controller-gen))
-	@{ \
-	set -e ;\
-	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.5 ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
-	}
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+${CONTROLLER_GEN}: | ${BIN}
+	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d)
+	trap "rm -rf $${CONTROLLER_GEN_TMP_DIR}" EXIT
+	cd $${CONTROLLER_GEN_TMP_DIR}
+	go mod init tmp
+	GOBIN=$${CONTROLLER_GEN_TMP_DIR} go get sigs.k8s.io/controller-tools/cmd/controller-gen@v${CONTROLLER_GEN_VERSION}
+	mv $${CONTROLLER_GEN_TMP_DIR}/controller-gen ${CONTROLLER_GEN}
+
```

Comment:
I install controller-gen in `./bin/controller-gen-0.5.0` along side the other build and test tools, like kind.
I think go 1.16 has a better `go install` mechanism which would not need all this tmpdir stuff:
 * https://golang.org/doc/go1.16#go-command